konnectivity-socks5-proxy/main.go

```diff
@@ -70,16 +71,19 @@ func NewStartCommand() *cobra.Command {
 			panic(err)
 		}
 
-		dialFunc := dialFunc(caCertPath, clientCertPath, clientKeyPath, proxyHostname, proxyPort, connectDirectlyToCloudAPIs)
+		fallbackToManagementCluster := new(bool)
+
+		dialFunc := dialFunc(caCertPath, clientCertPath, clientKeyPath, proxyHostname, proxyPort, connectDirectlyToCloudAPIs, fallbackToManagementCluster)
```

Comment:
why do we need fallbackToManagementCluster in the signature of dialFunc if it's always hardcoded to false?

Reply:
Is this because is changed at runtime by `(d proxyResolver) Resolve` via the `dnsFallback` pointer and then used by the next resolver relying the `dialFunc`?

If so it's worth a comment clarifying the above. Also The cyclomatic complexity for this component is becoming difficult to follow, I wonder how we could simplify

Reply:
Yeah it is because of what you said - changed at runtime by `(d proxyResolver) Resolve` via the `dnsFallback` pointer and then used by the next resolver relying the `dialFunc`. I will add a comment.